src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -47,54 +52,308 @@ export class PermissionsTableComponent implements OnInit {
+          .subscribe((roles: Role[]) => {
+            this.entities = this.getEntities(this.entityType, roles);
+            this.otherEntities = this.getOtherEntities(this.entityType, orgCode, roles);
+          });
+        });
+        break;
+      case 'project':
+        this.route.paramMap.subscribe((params: ParamMap) => {
+          let orgCode: string = params.get('organizationCode');
+          let proCode: string = params.get('projectCode');
+          this.rolesService.getProjectRoles(orgCode, proCode)
+          .subscribe((roles: Role[]) => {
+            this.entities = this.getEntities(this.entityType, roles);
+            this.otherEntities = this.getOtherEntities(this.entityType, orgCode, roles);
+          });
+        });
+    }
+  }
+
+  public updateCheckedRoles(role: string, event) {
+    this.rolesCheckbox[role] = event.target.checked;
+  }
+
+  public addOrganizationUsersGroupsRole(orgCode: string, role: string, users: string[], groups: string[]) {
+    this.rolesService.addOrganizationUsersGroupsRole(orgCode, role, users, groups)
+      .subscribe(
+      response => {;}
+    );
+  }
+
+  public removeOrganizationUsersGroupsRole(orgCode: string, role: string, users: string[], groups: string[]) {
+    this.rolesService.removeOrganizationUsersGroupsRole(orgCode, role, users, groups)
+      .subscribe(
+      response => {;}
+    );
+  }
+
+  public addProjectUsersGroupsRole(orgCode: string, projectCode: string, role: string, users: string[], groups: string[]) {
+    this.rolesService.addProjectUsersGroupsRole(orgCode, projectCode, role, users, groups)
+      .subscribe(
+      response => {;}
+    );
   }
 
-  private getEntities(entityType: string): Entity[] {
+  public removeProjectUsersGroupsRole(orgCode: string, projectCode: string, role: string, users: string[], groups: string[]) {
+    this.rolesService.removeProjectUsersGroupsRole(orgCode, projectCode, role, users, groups)
+      .subscribe(
+      response => {;}
+    );
+  }
+
+  public onAdd(name: string) {
+    let addedRoles = [];
+    switch (this.resourceType) {
+      case 'organization':
+        this.route.paramMap.subscribe((params: ParamMap) => {
+          let orgCode: string = params.get('organizationCode');
+          switch (this.entityType) {
+            case 'users':
+              for (let r in this.roles) {
+                if (this.rolesCheckbox[this.roles[r]] && this.roles.hasOwnProperty(r)) {
+                  this.addOrganizationUsersGroupsRole(orgCode, this.roles[r], [name], []);
+                  addedRoles.push(this.roles[r]);
+                }
+              }
+            break;
+            case 'groups':
+              for (let r in this.roles) {
+                if (this.rolesCheckbox[this.roles[r]] && this.roles.hasOwnProperty(r)) {
+                  this.addOrganizationUsersGroupsRole(orgCode, this.roles[r], [], [name]);
+                  addedRoles.push(this.roles[r]);
```

Comment:
It is not the same. Check the order of the parameters of the `addOrganizationUsersGroupsRole` function. In the first case, a role is added for a user in the second for a group.